applications/ChimeraApplication/custom_processes/apply_chimera_process.h

```diff
@@ -0,0 +1,391 @@
+
+namespace Kratos {
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template <int TDim>
+class KRATOS_API(CHIMERA_APPLICATION) ApplyChimera : public Process {
+public:
+    ///@name Type Definitions
+    ///@{
+
+    ///@}
+    ///@name Pointer Definitions
+    typedef Kratos::VariableComponent<Kratos::VectorComponentAdaptor<Kratos::array_1d<double, 3>>> VariableComponentType;
+    typedef ModelPart::IndexType IndexType;
+    typedef ModelPart::NodeType NodeType;
+    typedef Kratos::Variable<double> VariableType;
+    typedef std::vector<IndexType> ConstraintIdsVectorType;
+    typedef typename ModelPart::MasterSlaveConstraintType MasterSlaveConstraintType;
+    typedef typename ModelPart::MasterSlaveConstraintContainerType MasterSlaveConstraintContainerType;
+    typedef std::vector<MasterSlaveConstraintContainerType> MasterSlaveContainerVectorType;
+    typedef BinBasedFastPointLocator<TDim> PointLocatorType;
+    typedef typename PointLocatorType::Pointer PointLocatorPointerType;
+    typedef ModelPart::NodesContainerType NodesContainerType;
+
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyChimera);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Constructor
+     * @param rMainModelPart The reference to the modelpart which will be used
+     * for computations later on.
+     * @param iParameters The settings parameters.
+     */
+    explicit ApplyChimera(ModelPart& rMainModelPart, Parameters iParameters);
+
+    /// Destructor.
+    virtual ~ApplyChimera();
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void SetEchoLevel(int EchoLevel);
+
+    void SetReformulateEveryStep(bool Reformulate);
+
+    virtual void ExecuteInitializeSolutionStep() override;
+
+    virtual void ExecuteFinalizeSolutionStep() override;
+
+    virtual std::string Info() const override;
+
+    /// Print information about this object.
+    virtual void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    virtual void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+    ModelPart& mrMainModelPart;
+    int mNumberOfLevels; 
+    Parameters mParameters;
+    std::unordered_map<IndexType, ConstraintIdsVectorType> mNodeIdToConstraintIdsMap;
+    int mEchoLevel;
+    bool mReformulateEveryStep;
+    std::map<std::string, PointLocatorPointerType> mPointLocatorsMap;
+    bool mIsFormulated;
+
+    // Modelpart names which are generated here
+    std::string mModifiedName = "ChimeraModified";
+    std::string mBoundaryName = "ChimeraBoundary";
+    std::string mHoleName = "ChimeraHole";
```

Comment:
```suggestion
    const std::string mModifiedName = "ChimeraModified";
    const std::string mBoundaryName = "ChimeraBoundary";
    const std::string mHoleName = "ChimeraHole";
```
I don't know if this suggestion introduces any limitation but IMO it would make sense that these model parts have always the same name (specially when debugging).

Reply:
Ya but these are created as sub-modelparts of different other submodelparts. So the full name of the modelpart will always be different.